### PR TITLE
Diffuse mods

### PR DESCRIPTION
**Pull request by mewall**
These revisions are needed for consistency with the revisions to diffBragg

Note: There are some changes to the crystal rotation behavior, to allow for a faster sweep of reciprocal space. These can be backed out if there are objections.

### CODE REVIEW

File: command_line/sim_view.py [hunk 1]
```diff
@@ -526,12 +575,13 @@ def _toggle_Fhkl(self, _press=None):
         self._display()
 
     def _toggle_spectrum_shape(self, _press=None):
-        options = ["Gaussian", "SASE", "monochromatic"]
-        current = options.index(self.spectrum_shape)
-        self.spectrum_shape = options[(current+1)%3]
-        self._update_spectrum(init=(self.spectrum_shape == "SASE"))
-        self._generate_image_data()
-        self._display()
+        if not self.diffuse_scattering:
```
Comment (irisdyoung) [inline command_line/sim_view.py:578]: May be prudent to add a matching reset to monochromatic when diffuse is turned on.

File: command_line/sim_view.py [hunk 2]
```diff
@@ -808,14 +858,14 @@ def _reset(self, _press=None):
...
-        "Diff_aniso":[.01, 10, .01, .1, 2],
+        "Diff_gamma":[1, 300, 1, 10, 50],
+        "Diff_sigma":[.01, 0.7, .01, 0.05, .4],
+        "Diff_aniso":[.01, 10, .1, 1, 3],
         "Energy":[6500, 12000, 10, 30, 9500],
         "Bandwidth":[0.01, 5.01, 0.1, 1, 0.31],
-        "RotX": [-180, 180, 0.01, 0.1, 0],
-        "RotY": [-180, 180, 0.01, 0.1, 0],
-        "RotZ": [-180, 180, 0.01, 0.1, 0],
+        "RotX": [-180, 180, 0.1, 1., 0],
```
Comment (irisdyoung) [inline command_line/sim_view.py:866]: For SFX purposes I think we do care about misorientations of 0.01 degrees and would want to keep the small step size this small. Open to negotiation on the big step size. @nksauter please contradict me if this doesn't match your understanding.

**Comment by mewall** [comment]
Does the block starting at 560 address this sufficiently?

________________________________
From: Iris Young ***@***.***>
Sent: Wednesday, September 7, 2022 1:08:38 PM
To: dermen/sim_erice
Cc: Wall, Michael E; Author
Subject: [EXTERNAL] Re: [dermen/sim_erice] Diffuse mods (PR #7)


@irisdyoung commented on this pull request.

________________________________

In command_line/sim_view.py<https://urldefense.com/v3/__https://github.com/dermen/sim_erice/pull/7*discussion_r965184482__;Iw!!Bt8fGhp8LhKGRg!EZ5pkI4gJBK3YpnQemjOLwSgLyK38VB-ZPg15LSkWnBBxKioWYRo54FqhW49WlAzUbWP5on6VznfPfyqAfzOK2Hq$>:

> @@ -526,12 +575,13 @@ def _toggle_Fhkl(self, _press=None):
         self._display()

     def _toggle_spectrum_shape(self, _press=None):
-        options = ["Gaussian", "SASE", "monochromatic"]
-        current = options.index(self.spectrum_shape)
-        self.spectrum_shape = options[(current+1)%3]
-        self._update_spectrum(init=(self.spectrum_shape == "SASE"))
-        self._generate_image_data()
-        self._display()
+        if not self.diffuse_scattering:


May be prudent to add a matching reset to monochromatic when diffuse is turned on.

**Comment by mewall** [comment]
I thought you might want to keep that.


I think 0.1 is fine for the diffuse scattering mode, so maybe we can just change these parameters when switching to that mode, and change them back when exiting. Would that be a good fix?


________________________________
From: Iris Young ***@***.***>
Sent: Wednesday, September 7, 2022 1:11:57 PM
To: dermen/sim_erice
Cc: Wall, Michael E; Author
Subject: [EXTERNAL] Re: [dermen/sim_erice] Diffuse mods (PR #7)


@irisdyoung commented on this pull request.

________________________________

In command_line/sim_view.py<https://urldefense.com/v3/__https://github.com/dermen/sim_erice/pull/7*discussion_r965186935__;Iw!!Bt8fGhp8LhKGRg!EjMAkBm15bSywtzcJtjpC-H0L35HWNOHqDdHAP61imE5GPSrczHb-xsPcZUjvv-6CWal13gSj9X5Atnu_MT6WA4F$>:

>          "Energy":[6500, 12000, 10, 30, 9500],
         "Bandwidth":[0.01, 5.01, 0.1, 1, 0.31],
-        "RotX": [-180, 180, 0.01, 0.1, 0],
-        "RotY": [-180, 180, 0.01, 0.1, 0],
-        "RotZ": [-180, 180, 0.01, 0.1, 0],
+        "RotX": [-180, 180, 0.1, 1., 0],


For SFX purposes I think we do care about misorientations of 0.01 degrees and would want to keep the small step size this small. Open to negotiation on the big step size. @nksauter<https://urldefense.com/v3/__https://github.com/nksauter__;!!Bt8fGhp8LhKGRg!EjMAkBm15bSywtzcJtjpC-H0L35HWNOHqDdHAP61imE5GPSrczHb-xsPcZUjvv-6CWal13gSj9X5Atnu_PycnY_a$> please contradict me if this doesn't match your understanding.

**Comment by irisdyoung** [comment]
Re: monochromatic mode, yes that does address it. I missed that the first time through. 

Re: step size, that sounds good for now, and I will make it a user-adjustable parameter in another round of changes a little later.

**Comment by mewall** [comment]
@irisdyoung OK now?

> Re: monochromatic mode, yes that does address it. I missed that the first time through.
> 
> Re: step size, that sounds good for now, and I will make it a user-adjustable parameter in another round of changes a little later.

**Pull request merged**